packages/rum-core/src/performance-monitoring/capture-hard-navigation.js

```diff
@@ -232,75 +232,81 @@ function createUserTimingSpans(entries, transactionEnd) {
+   * Threshold that decides if the span must be
+   * captured as part of the page load transaction
+   *
+   * Denotes the time when the onload event fires
+   */
+  const transactionEnd = transaction._end
+  const timings = perf.timing
 
+  createNavigationTimingSpans(
+    timings,
+    timings.fetchStart,
+    transactionEnd
+  ).forEach(span => {
+    span.traceId = transaction.traceId
+    span.sampled = transaction.sampled
+    if (span.pageResponse && transaction.options.pageLoadSpanId) {
+      span.id = transaction.options.pageLoadSpanId
+    }
+    transaction.spans.push(span)
+  })
+
+  if (typeof perf.getEntriesByType === 'function') {
     /**
      * capture resource timing information as Spans during page load transaction
      */
-    if (typeof perf.getEntriesByType === 'function') {
-      const resourceEntries = perf.getEntriesByType('resource')
+    const resourceEntries = perf.getEntriesByType('resource')
 
-      const ajaxUrls = []
-      for (let i = 0; i < transaction.spans; i++) {
-        const span = transaction.spans[i]
+    const ajaxUrls = []
+    for (let i = 0; i < transaction.spans; i++) {
+      const span = transaction.spans[i]
 
-        if (span.type === 'external' && span.subType === 'http') {
-          continue
-        }
-        ajaxUrls.push(span.name.split(' ')[1])
+      if (span.type === 'external' && span.subType === 'http') {
+        continue
       }
-      createResourceTimingSpans(
-        resourceEntries,
-        ajaxUrls,
-        transactionEnd
-      ).forEach(span => transaction.spans.push(span))
+      ajaxUrls.push(span.name.split(' ')[1])
+    }
+    createResourceTimingSpans(
+      resourceEntries,
+      ajaxUrls,
+      transactionEnd
+    ).forEach(span => transaction.spans.push(span))
 
-      const userEntries = perf.getEntriesByType('measure')
-      createUserTimingSpans(userEntries, transactionEnd).forEach(span =>
-        transaction.spans.push(span)
-      )
+    /**
+     * capture user timing information as spans
+     */
+    const userEntries = perf.getEntriesByType('measure')
+    createUserTimingSpans(userEntries, transactionEnd).forEach(span =>
+      transaction.spans.push(span)
+    )
 
-      /**
-       * Add transaction context information from performance navigation timing entry level 2 API
-       */
-      let navigationEntry = perf.getEntriesByType('navigation')
-      if (navigationEntry && navigationEntry.length > 0) {
-        navigationEntry = navigationEntry[0]
-        transaction.addContext({
-          response: getResponseContext(navigationEntry)
-        })
-      }
+    /**
+     * Add transaction context information from performance navigation timing entry level 2 API
+     */
+    let navigationEntry = perf.getEntriesByType('navigation')
+    if (navigationEntry && navigationEntry.length > 0) {
+      navigationEntry = navigationEntry[0]
+      transaction.addContext({
+        response: getResponseContext(navigationEntry)
+      })
     }
```

Comment:
The changes in the capture-hard-navigation.js seems to be only moving the code around, is that correct? or am I missing something?